src/inference/graph_inference.cpp

```diff
@@ -341,6 +342,24 @@ class GraphNodeAssignment : public Nice2Assignment {
     return label_set_->GetLabelName(label_id);
   }
 
+  // Returns conditional probability of a label conditionated on its neighbouring labels
```

Comment:
Given that GetNodeNormalizationConstantGivenNeighbours has some "hidden" functionality with appending to the vector, I am not convinced it is a good idea to have the methods. You can have the methods if you can clearly describe what they do. The first on is just a formula that you call once, the second one you cannot describe even with the super long name.